### ISSUE TITLE
Unable to create a JWS with a private key that doesn't include dp, dq, qi

### ISSUE THREAD

**Issue by nubs** [Spomky-Labs/jose#111]
This may not be the right place to report this, so I would like to apologize ahead of time if that's the case.

I'm trying to sign a JWS, but the private key that I have that comes from a JWK api endpoint does not have dp, dq, or qi and that seems to cause issues.  I have some failing example code [here](https://gist.github.com/nubs/e5e1f67904ec4ee5a192511a1d0fb4b7) with an example private key.

I'm basing my work off of a provided example javascript application that seems to have had a similar issue, but was able to get around it by setting those three values to empty strings (https://github.com/ory-am/hydra-idp-react/blob/master/src/common/service/hydra.js#L114-L128).  Attempting to do that here does succeed in creating a JWS, but that JWS's signature won't verify afterwards.

**Comment by Spomky** [comment]
I am not sure if it is the right place or not.
The library already set 0 for all missing exponents (see [these lines](https://github.com/Spomky-Labs/jose/blob/master/src/KeyConverter/RSAKey.php#L103-L105)).
The issue might come from a dependency (PHPSecLib), but it seems that it is [already covered by the library](https://github.com/phpseclib/phpseclib/blob/master/phpseclib/Crypt/RSA.php#L1302-L1310).

I am trying to figure out where the issue comes from.
Stay tuned.

**Comment by nubs** [comment]
Awesome, thanks

**Comment by Spomky** [comment]
The problem definitevly comes from the PHPSecLib.
I updated the library to use openssl instead of that library for all `RSxxx` algorithms.

The problem still exist when using `PSxxx` algorithms and all RSA based key encryption algorithms (`RSA1_5`, `RSA-OAEP` and `RSA-OAEP-256`).
You can report that bug on the PHPSecLib project, however I think I will get rid of it and use openssl or an homemade solution.

**Comment by nubs** [comment]
Awesome, I'll follow up over there.  Thanks for all of your help yesterday and today.

**Comment by Spomky** [comment]
Tagged `5.0.2`.

You are welcome.
I opened #112

**Comment by nubs** [comment]
It works now, no problem.  I'll try to follow up and create an issue in the next few days for getting it fixed in phpseclib.  You're the best!

**Comment by Spomky** [comment]
Hi Spencer,

I am very happy to announce you that I solved the issue for all algorithms.
The bug seems to be completely resolved, however I need to perform some code design works before tagging a new version.

I also removed the PHPSecLib and now the library uses its own implementation.

Just a question: how did you generate such a key without additional primes? I would like to create new keys without that primes.
